rpc/gnmi/gnmi-union_replace.md

```diff
@@ -0,0 +1,327 @@
+
+### 5.3 Union behavior options
+
+This section describes how the union between native and OC should be performed.  Different scenarios are described for handling CLI vs. Native origin union with OpenConfig.
+In addition, there are two options for handling errors in the union process. A NOS must assert which error handling option is implemented in it's documentation.
+
+#### 5.3.1 union behavior for CLI and OpenConfig
+
+As the origins are joined together, the precedence specified (per origin) should be followed to determine the order in which they are applied. The concept of precedence is defined in bootz.
+
+bootz defines CLI to be precedence 100, and `origin: openconfig` to be precedence 110.  Configuration is merged in the reverse precedence order.  For example,  precedence 0 should be merged last, overriding any higher precedence value.  A Set operation specifying `union_replace` for both CLI and OC origins should be performed as follows:
+
+1. Create a candidate configuration using factory defaults.
+   1. Factory defaults are used because CLI is expected to represent the entire configuration, subject to any partitioning rules of bootz and gNSI.
+2. Process OC paths, replacing the relevant configuration items in the candidate  configuration.  This includes OC default values.
+3. Process CLI items merging them to the candidate configuration.  Generate an error if there is conflict between OC and CLI as defined below.  Conflict and resolution is defined in [5.3.3](5.3.3-Resolving-issues-with-union-between-the-origins).
+
+#### 5.3.2 union behavior for Native and OpenConfig
+
+We define a native origin to be precedence 100, and `origin: openconfig` to be precedence 110.  A Set operation specifying `union_replace` operation for both native and `openconfig` origins MUST be performed as follows. Additional consideration is required because the native yang for a device is (by definition) modeled and hence the path in the message must be considered.
+
+1. Create a candidate config using the current running configuration.
+   1. Running configuration is used because the NY paths may be a subtree.
+2. Process OC paths, replacing the relevant subtree(s) in the candidate configuration.  This includes OC default values.
+3. Process native paths, replacing the relevant subtree(s) in the candidate configuration.  Generate an error if there is conflict between OC and Native as defined below.
+
+#### 5.3.3 Resolving issues with union between the origins
+
+##### 5.3.3.1 Default values
+
+The following rules govern how default values are applied when using two origins.
+
+1. Defaults from OC apply at the container level.
```

Comment:
What does this mean?

Reply:
I will reference the yang specification for default applications.